File: MATTHEW/leetcode/AIcoding/4.GPT-4o/cpp_solutions/3513-number-of-unique-xor-triplets-i.cpp
class Solution {
public:
    int countTriplets(vector<int>& nums) {
        int n = nums.size();
        int count = 0;
        unordered_set<int> uniqueXORs;

        for (int i = 0; i < n; ++i) {
            for (int j = i + 1; j < n; ++j) {
                for (int k = j + 1; k < n; ++k) {
                    int xorValue = nums[i] ^ nums[j] ^ nums[k];
                    uniqueXORs.insert(xorValue);
                }
            }
        }

        return uniqueXORs.size();
    }
};

File: leetcode/AIcoding/1.Gemini-2.5/javascript_solutions/3513-number-of-unique-xor-triplets-i.js
var countTriplets = function(nums) {
    let count = 0;
    const n = nums.length;

    for (let i = 0; i < n; i++) {
        for (let j = i + 1; j < n; j++) {
            for (let k = j + 1; k < n; k++) {
                if ((nums[i] ^ nums[j] ^ nums[k]) === 0) {
                    count++;
                }
            }
        }
    }

    return count;
};